database/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,75 @@
+package neo4j
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/dhui/dktest"
+	"github.com/neo4j/neo4j-go-driver/neo4j"
+
+	dt "github.com/golang-migrate/migrate/v4/database/testing"
+	"github.com/golang-migrate/migrate/v4/dktesting"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+)
+
+var (
+	opts = dktest.Options{PortRequired: true, ReadyFunc: isReady}
+	specs = []dktesting.ContainerSpec{
+		{ImageName: "neo4j:3.5", Options: opts},
+		//{ImageName: "neo4j:3.5-enterprise", Options: opts},
+	}
+)
+
+func neoConnectionString(host, port string) string {
+	return fmt.Sprintf("bolt://neo4j:neo4j@%s:%s", host, port)
+}
+
+func isReady(ctx context.Context, c dktest.ContainerInfo) bool {
+	ip, port, err := c.Port(7687)
```

Comment:
hardcoded port?

Reply:
This is for a test, the function asks the docker test suite which docker port has been mapped to 7687, which is the bolt exposed port in the neo docker container. In real usage, the port would be provided by the URL the user gives